app/models/WhatDoYouWantToDo.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.data.Form
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.Aliases.Text
+import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.RadioItem
+import views.ViewUtils.hint
+
+sealed trait WhatDoYouWantToDo
+
+object WhatDoYouWantToDo extends Enumerable.Implicits[WhatDoYouWantToDo] {
+
+  case object CreateOption extends WithName("createOption") with WhatDoYouWantToDo
+
+  case object UpdateOption extends WithName("updateOption") with WhatDoYouWantToDo
+
+  case object CancelOption extends WithName("cancelOption") with WhatDoYouWantToDo
```

Comment:
Could these just be "CreateCase", "UpdateCase", "CancelCase" instead of Option

Reply:
agree

Reply:
I'm not sure if I like the `WhatDoYouWantToDo` name too...
I think something like `UserFlow`, `SubmissionType` or something like that would be better